src/components/Forms/PhoneNumberField.js

```diff
@@ -0,0 +1,300 @@
+    /** Disable the text field */
+    disabled           : PropTypes.bool,
+    /** Text of label that will animate when TextField is focused */
+    floatingLabelText  : PropTypes.string,
+    /** Sets width to 100% */
+    fullWidth          : PropTypes.bool,
+    /** Sets width to 162px */
+    halfWidth          : PropTypes.bool,
+    /** FormComponent error for validation */
+    hasError           : PropTypes.bool,
+    /** Helper text will show up in bottom right corner below TextField */
+    helperText         : PropTypes.string,
+    /** Uniq id for input */
+    id                 : PropTypes.string,
+    /** Style for input */
+    inputStyle         : PropTypes.object,
+    /** Set by FormComponent by default.   */
+    isValid            : PropTypes.bool,
+    /** onFocus callback */
+    onFocus            : PropTypes.func,
+    /** onChange callback */
+    onChange           : PropTypes.func,
+    /** onBlur callback */
+    onBlur             : PropTypes.func,
+    /** onKeyDown callback */
+    onKeyDown          : PropTypes.func,
+    /** Mark the field as required.  */
+    required           : PropTypes.bool,
+    /** Error from server to show ServerError message */
+    serverError        : PropTypes.string,
+    /** Wrapper styles */
+    style              : PropTypes.object,
+    /** Text to show for validation error  */
+    validationErrorText: PropTypes.string,
+    /** Value will make TextField a controlled component  */
+    value              : PropTypes.string,
+    /** Snacks theme attributes provided by `Themer` */
+    snacksTheme        : themePropTypes
+  }
+
+  static defaultProps = {
+    autoComplete: 'on',
+    disabled    : false,
+    defaultValue: null,
+    onKeyDown: () => {} // eslint-disable-line no-empty-function
+  }
+
+  state = {
+    hasValue: this.props.defaultValue !== null || Boolean(this.props.value)
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.disabled && !this.props.disabled) {
+      this.setState({ isFocused: false })
+    }
+
+    if (!this.state.hasValue && nextProps.value) {
+      this.setState({hasValue: true})
+    }
+  }
+
+  getValue = () => {
+    if (!this.input) {
+      return null
+    }
+
+    return this.input.value.replace(/\(|\)|-| /g, '')
+  }
+
+  handleInputChange = (e) => {
+    const { onChange } = this.props
+    const { hasValue } = this.state
+    const value        = e.target.value
+    // Limit setState call to only when hasValue changes
+    if (value && !hasValue) {
+      this.setState({hasValue: true})
+    } else if (!value && hasValue) {
+      this.setState({hasValue: false})
+    }
+
+    onChange && onChange(e, value.replace(/\(|\)|-| /g, ''), value)
+  }
+
+  handleInputFocus = (e) => {
+    this.setState({isFocused: true})
+    this.props.onFocus && this.props.onFocus(e)
+  }
+
+  handleInputBlur = (e) => {
+    this.setState({isFocused: false})
+    this.props.onBlur && this.props.onBlur(e)
```

Comment:
nit - mostly personal preference but I like giving noops as a default value for optional callback props like this (which means lightly less code as you don't have to check that they exist).